besu/src/main/java/org/hyperledger/besu/controller/IbftBesuControllerBuilder.java

```diff
@@ -141,14 +141,14 @@ protected MiningCoordinator createMiningCoordinator(
         BftExecutors.create(metricsSystem, BftExecutors.ConsensusType.IBFT);
 
     final Address localAddress = Util.publicKeyToAddress(nodeKey.getPublicKey());
-    final BftBlockCreatorFactory blockCreatorFactory =
-        new BftBlockCreatorFactory(
+    final BftBlockCreatorFactory<?> blockCreatorFactory =
+        new BftBlockCreatorFactory<>(
             transactionPool.getPendingTransactions(),
             protocolContext,
             protocolSchedule,
+            forksSchedule,
```

Comment:
Pass `forkSchedule` to block creator so we can calculate the mining beneficiary based on the configured forks.